### PR TITLE
Support RFC 850 date format #696

### PR DESCRIPTION
**Pull request by lempere**

### CODE REVIEW

File: ktor-utils/src/io/ktor/util/date/ParseDate.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    return try {
+        val subDate = rfc850.subSequence(rfc850.indexOf(',')+2, rfc850.length)
+        val parts = subDate.split(' ')
+        val date = parts[0].split('-')
+        val time = parts[1].split(':')
+
+        GMTDate(time[2].toInt(),time[1].toInt(),time[0].toInt(), date[0].toInt(), Month.from(date[1]), yearCommonEra( date[2].toInt() ) )
+    }catch (ex:Exception){
+        GMTDate.START
```
Comment (e5l) [inline ktor-utils/src/io/ktor/util/date/ParseDate.kt]: Maybe we should throw `FailToParseDateException` here?

File: ktor-utils/src/io/ktor/util/date/ParseDate.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+}
+
+fun yearCommonEra(year: Int) = if (year<70) year + 2000 else year + 1900
```
Comment (e5l) [inline ktor-utils/src/io/ktor/util/date/ParseDate.kt]: private or internal

**Comment by e5l** [comment]
Hi @lempere, thanks for the PR.
Could you use `*` imports everywhere as in ktor code style?
And probably enable autoformatting to fix minor formatting issues.

File: ktor-utils/src/io/ktor/util/date/ParseDate.kt (hunk 1 above)
Comment (lempere) [reply ktor-utils/src/io/ktor/util/date/ParseDate.kt]: Or maybe null could do the job

Comment (e5l) [reply ktor-utils/src/io/ktor/util/date/ParseDate.kt]: `null` is for missing

**Pull request closed without merging**